requirements.txt

```diff
@@ -1,6 +1,7 @@
 # the default package dependencies
 
 numpy>=1.16.6
+matplotlib>3.1
```

Comment:
please move it to extras or loggers
it should not be a base requirement

Reply:
This would break the type annotations though wouldn't it?

Reply:
yes it would, but I think it's better to use `Any` for typing then having this as a hard requirement here.

also you could do something like this:

```python
if _MATPLOTLIB_AVAILABLE:
    from matplotlib import pyplot as plt
else:
    class plt:
        figure = None
        close = None
```

You would then also have to include checks and raise proper warnings inside the functions. But for such an optional feature we usually don't want hard requirements.

Reply:
Hmm I thought there is hardly any environment without matplotlib 😅  (but I might be biased)

@justusschock Okay, if I opt for the `Any` solution that means that all log_figure() methods must include a 
local `import matplotlib.pyplot as plt` instead of at top level.

i.e. 
```python
@rank_zero_only
def log_figure(self, name: str, figure, step: Optional[int] = None, close: bool = True) -> None:
    import matplotlib.pyplot as plt
    # ...
    plt.close(figure)
```

there is no real need to check / issue warnings once I am inside the method because if a plt.figure is passed the user will have matplotlib installed, right?